### DISCUSSION TITLE
clickhouse offline provider setup help

### DISCUSSION THREAD

**Discussion by umeshkacha** [featureform/featureform#1537] [Q&A]
I am trying to use clickhouse as offline provider of featureform. I have the following code taken from ClickHouse [documentation](https://clickhouse.com/blog/training-machine-learning-models-with-clickhouse-and-featureform) but it gives error saying Internal Error: invalid config update.

```
from featureform import Client
import featureform as ff

# Featureform client. Local instance and insecure
client = Client(insecure=True)
# Register our local container with Featureform
clickhouse = ff.register_clickhouse(
   name="clickhouse",
   description="A ClickHouse deployment for example",
   host="quickstart-clikchouse",
   port=8123,
   user="default",
   password="",
   database="fraud"
)
client.apply(verbose=True)
```

**Comment by epps** [comment]
@umeshkacha, it appears you have a spelling error in the host field: `"quickstart-clikchouse"` should be `"quickstart-clickhouse"`. Given Featureform disallows updating the host for a ClickHouse provider, this is likely why you're seeing this issue. But this would only occur if you were trying to register the provider a second time. Is that the case here?

One other thing I noticed: our quickstart is using `clickhouse/clickhouse-server` as the image; however, it really should be `featureform/clickhouse-server` is the image that contains the preloaded fraud database. I would kill your currently running ClickHouse container and run `featureform/clickhouse-server` instead if you're planning on working through the rest of the blog post.

Let me know if this resolves your issue.

**Comment by umeshkacha** [comment]
@epps Did you mean docker pull featureformcom/clickhouse-server? If yes I have the same image. If not then docker pull featureform/clickhouse-server says no repo found so I guess no docker image for that. On a side not for above example what do you set as FEATUREFORM_HOST I have set localhost:7878
